improve visibility of bug reporting guidelines.

vaidap opened the pull request:
Fixes #7388.

Thanks to @cg-cnu for the link to a basic  `issue_template.md`!

The points left to cover are:
* Adapting `issue_template.md` to fit Zulip specifically
* Linking to examples of good bug reports in the guidelines
* Giving the new contributor empathy for the receiver of the bug report, by possibly emphasising the relevant section of https://www.chiark.greenend.org.uk/~sgtatham/bugs.html.

I'd appreciate links/feedback on how to achieve this. :)

vaidap commented:
good point! 
* changed `issue-template.md` to only have a link to the bug report guidelines
* added paragraph about understanding why bug fixing can be a frustrating process.

rishig commented:
Thanks for talking a stab at this @vaidap! 
I'm tempted to say this is a pretty difficult issue to tackle, and may just require writing by someone who has dealt with the other end of this directly.

The issue is not that the person reporting bugs gets tired of answering questions, it's that the person triaging bug reports gets tired of asking the same question to lots of people.

vaidap commented:
Ah, my mistake! I've removed that paragraph, and added a couple of bug reports that are well-written. I won't add any more changes & let someone with more experience with receiving reports add to the document. :)

timabbott commented:
Closing, since I've been pretty happy with the quality of bug reports that we've been receiving organically without such a page.

Thanks for working on this @vaidap!
